### DISCUSSION TITLE
Importing formatted names does not work correctly

### DISCUSSION THREAD

**Discussion by Michmcb** [you-apps/ConnectYou#422] [General]
### Steps to reproduce

When importing a vcard, the formatted name property is not imported correctly
[Anthony Hickey.export.vcf.txt](https://github.com/user-attachments/files/17567035/Anthony.Hickey.export.vcf.txt)
[Anthony Hickey.import.vcf.txt](https://github.com/user-attachments/files/17567036/Anthony.Hickey.import.vcf.txt)

### Expected behavior

When importing Anthony Hickey with formatted name Tony, that data should be imported as is

### Actual behavior

The name is incorrectly made Tony. That should be set as the nickname

### Connect You version

9.0 (28)

### Android version

Android 13

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
Your VCard contains `Tony` as the formatted name field as per https://www.rfc-editor.org/rfc/rfc6350.html#section-6.2.1.

Hence the app sets `Tony` as the full name, which is the expected behavior.

If `Tony` should have only been a nickname, `Tony` should be in the `NICKNAME` field as per https://www.rfc-editor.org/rfc/rfc6350.html#section-6.2.3.

**Reply by Michmcb** [reply]
Shouldn't the full name on import be set from the `N` field, which specifies each component of the name? Whereas `FN` is formatted text corresponding to name of the object the vCard represents, which may not necessarily be identical to the `N` components concatenated together.
Fair enough that the Nickname box in the app corresponds to the optional property `NICKNAME` in the RFC, what does `FN` correspond to then?

**Reply by Michmcb** [reply]
In addition, what if the name order differs? For example, if `FN` has a `LANGUAGE` parameter, and that culture formats names in the order Family, Given? Shouldn't the import use the `N` property which explicitly specifies which parts are Family, Given?

**Reply by Michmcb** [reply]
Or another example; say the Dutch name "Pieter van der Meer". Although it is formatted that way, "van der" is the prefix. So the `N` property would be `N:Meer;Pieter;;van der;` but the correct way to write/format the name is `Pieter van der Meer`.

**Reply by Bnyro** [reply]
Connect You doesn't do the VCard name parsing itself, we use the [`ezvcard`](https://github.com/mangstadt/ez-vcard) library, I can't tell you how it works exactly.

All we do is https://github.com/you-apps/ConnectYou/blob/main/app/src/main/java/com/bnyro/contacts/util/VcardHelper.kt#L105-L107.

**Reply by Michmcb** [reply]
Right, I'll have a read of the code, and probably raise an issue on that repo. Thanks for the information!
